From b32678a349ec3e8a526180992bfe9a72f121fd9d Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Tue, 2 Mar 2021 11:06:37 +0200
Subject: [PATCH 54/70] iot-gate-imx8: dts: add M4 support

The RPMsg is used for communication between the Cortex-A53 and the
Cortex-M4 cores.
Enable the RPMsg.

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/Makefile              |  1 +
 .../arm64/boot/dts/compulab/sb-iotgimx8-ied-m4.dts | 57 ++++++++++++++++++++++
 2 files changed, 58 insertions(+)
 create mode 100644 arch/arm64/boot/dts/compulab/sb-iotgimx8-ied-m4.dts

diff --git a/arch/arm64/boot/dts/compulab/Makefile b/arch/arm64/boot/dts/compulab/Makefile
index 4f5d631670b2..4a2ce1347f8c 100644
--- a/arch/arm64/boot/dts/compulab/Makefile
+++ b/arch/arm64/boot/dts/compulab/Makefile
@@ -6,6 +6,7 @@ dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-mini.dtb
 dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8.dtb
 dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8-can.dtb
 dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8-ied.dtb
+dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8-ied-m4.dtb
 dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8-poed.dtb
 
 always         := $(dtb-y)
diff --git a/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied-m4.dts b/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied-m4.dts
new file mode 100644
index 000000000000..14ee90e77b02
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied-m4.dts
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2021 CompuLab Ltd.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ */
+
+#include "sb-iotgimx8-ied.dts"
+
+/ {
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		m4_reserved: m4@0x80000000 {
+			no-map;
+			reg = <0 0x80000000 0 0x1000000>;
+		};
+
+		vdev0vring0: vdev0vring0@b8000000 {
+			compatible = "shared-dma-pool";
+			reg = <0 0xb8000000 0 0x8000>;
+			no-map;
+		};
+
+		vdev0vring1: vdev0vring1@b8008000 {
+			compatible = "shared-dma-pool";
+			reg = <0 0xb8008000 0 0x8000>;
+			no-map;
+		};
+
+		vdevbuffer: vdevbuffer@b8400000 {
+			compatible = "shared-dma-pool";
+			reg = <0 0xb8400000 0 0x100000>;
+			no-map;
+		};
+	};
+};
+
+&rpmsg{
+	/*
+	 * 64K for one rpmsg instance:
+	 * --0xb8000000~0xb800ffff: pingpong
+	 */
+	vdev-nums = <1>;
+	reg = <0x0 0xb8000000 0x0 0x10000>;
+	memory-region = <&vdevbuffer>;
+	status = "okay";
+};
-- 
2.11.0

